fe/fe-core/src/main/java/org/apache/doris/datasource/property/storage/S3PropertyUtils.java

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.property.storage;
+
+import org.apache.doris.common.UserException;
+import org.apache.doris.common.util.S3URI;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
+
+public class S3PropertyUtils {
+
+
+    private static final String URI_KEY = "uri";
+
+    public static String constructEndpointFromUrl(Map<String, String> props,
+                                                  String stringUsePathStyle, String stringForceParsingByStandardUri) {
+        String uri = props.get(URI_KEY);
+        if (uri == null || uri.isEmpty()) {
+            return null;
+        }
+        boolean usePathStyle = Boolean.parseBoolean(stringUsePathStyle);
+        boolean forceParsingByStandardUri = Boolean.parseBoolean(stringForceParsingByStandardUri);
+        try {
+            S3URI s3uri = S3URI.create(uri, usePathStyle, forceParsingByStandardUri);
+            return s3uri.getEndpoint().orElse(null);
+        } catch (UserException e) {
+            return null;
+        }
+    }
+
+    public static String constructRegionFromUrl(Map<String, String> props, String stringUsePathStyle,
```

Comment:
Why not use `boolean` type for `UsePathStyle` and `ForceParsingByStandardUri`?
And you did not test this method with `UsePathStyle` and `ForceParsingByStandardUri` set to true